puppet/modules/redmine/manifests/init.pp

```diff
@@ -1,13 +1,34 @@
 # Hacky class to set up Redmine
 # Needs VCS handling for cloning and bundling Redmine itself
 #
-class redmine {
-  $app_root    = '/usr/share/redmine'
-  $db_name     = 'redmine4'
-  $db_username = 'adminpz8bn8d'
-  $db_password = cache_data('foreman_cache_data', 'db_password', random_password(32))
-  $password    = postgresql_password($db_username, $db_password)
-
+# === Parameters:
+#
+# $secret_token::   Token used in the Rails initializer for session auth, etc
+#
+# $email_password:: Mailgun SMTP access password
+#
+# $app_root::       Directory holding the application
+#
+# $data_dir::       Directory containing the data
+#
+# $username::       User to run under
+#
+# $db_name::        Name of the database
+#
+# $db_password::    The password used to connect to redmine
+#
+# $https::          Whether to enable the https vhost
+#
+class redmine (
+  String $secret_token           = 'token',
+  String $email_password         = 'pass',
+  Stdlib::Absolutepath $app_root = '/usr/share/redmine',
+  Stdlib::Absolutepath $data_dir = '/usr/share/redmine_data',
```

Comment:
Set this to where we really want it, I can set a temporary override in Foreman for now.